src/include/types.h

```diff
@@ -567,30 +567,38 @@ __s32  hostos_to_ceph_errno(__s32 e);
 #endif
 
 struct errorcode32_t {
-  int32_t code;
+  __s32 code;
```

Comment:
All internal error codes (e.g. xxx_errno.cc ) use `__s32`. I am not sure this is correct type since it comes from kernel, but I just wanted to have the same type being used for all error codes related stuff. Eventually these 2 types are the same. I am fine to use more C++ way you suggested of `using code_t = __s32` to "distinguesh" between kernel and user space types